services/gif/gif.go

```diff
@@ -0,0 +1,108 @@
+package gif
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/status-im/status-go/multiaccounts/accounts"
+)
+
+var tenorAPIKey = ""
+var defaultParams = "&media_filter=minimal&limit=50&key="
+
+const maxRetry = 3
+const baseURL = "https://g.tenor.com/v1/"
+
+func NewGifAPI(db *accounts.Database) *API {
+	return &API{db, &http.Client{Timeout: time.Minute}}
+}
+
+// API is class with methods available over RPC.
+type API struct {
+	db         *accounts.Database
+	httpClient *http.Client
+}
+
+func (api *API) SetTenorAPIKey(key string) (err error) {
+	log.Info("[GifAPI::SetTenorAPIKey]")
+	err = api.db.SaveSetting("gifs/api-key", key)
+	if err != nil {
+		return err
+	}
+	tenorAPIKey = key
+	return nil
+}
+
+func (api *API) GetContentWithRetry(path string) (value string, err error) {
+	var currentRetry = 0
+	var response *http.Response
+
+	for currentRetry < maxRetry {
+		response, err = api.httpClient.Get(baseURL + path + defaultParams + tenorAPIKey)
+		if err != nil {
+			log.Error("can't get content from path %s", path)
+			currentRetry++
+			time.Sleep(100 * time.Millisecond)
+		} else {
+			break
+		}
+	}
+
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("Status error: %v", response.StatusCode)
+	}
+
+	data, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return "", fmt.Errorf("Read body: %v", err)
+	}
+
+	return string(data), nil
+}
+
+func (api *API) FetchGifs(path string) (value string, err error) {
+	log.Info("[GifAPI::fetchGifs]")
+	return api.GetContentWithRetry(path)
+}
+
+func (api *API) UpdateRecentGifs(updatedGifs json.RawMessage) (err error) {
```

Comment:
rather than RawMessage, could we have typed struct here?

Reply:
Done and for unit tests I have created a different task so that this lands in the meantime :)
https://github.com/status-im/status-go/issues/2508